subworkflows/local/bam_variant_calling_haplotypecaller/main.nf

```diff
@@ -103,6 +117,8 @@ workflow BAM_VARIANT_CALLING_HAPLOTYPECALLER {
 
     // add variantcaller to meta map and remove no longer necessary field: num_intervals
     vcf = vcf.map{ meta, vcf -> [ meta - meta.subMap('num_intervals') + [ variantcaller:'haplotypecaller' ], vcf ] }
+    // TO-DO: Figure out whether it might be sufficient to add the variantcaller tag in an "input" channel eariler in the script?
```

Comment:
good idea

Reply:
How about I add `variantcaller:'haplotypecaller'` in `cram_intervals` at the very top of that script?
```
// Combine cram and intervals for spread and gather strategy
    cram_intervals = cram.combine(intervals)
        // Move num_intervals to meta map
        .map{ meta, cram, crai, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals, variantcaller:'haplotypecaller' ], cram, crai, intervals, [] ] }

    GATK4_HAPLOTYPECALLER(cram_intervals, fasta, fasta_fai, dict.map{ meta, dict -> [ dict ] }, dbsnp, dbsnp_tbi)
```
It seems to me that the addition of the `variantcaller` in the `meta` of the channel `cram_intervals` should "propagate" to all the other `meta`'s in that script.

The following pytest passed locally: `pytest --tag variant_calling --tag haplotypecaller`